### DISCUSSION TITLE
Prevent building the page in run time when fallback is set to true

### DISCUSSION THREAD

**Discussion by MohammadPCh** [vercel/next.js#12613] [Help]
Think About situation you set fallback true in getStaticPaths, for exmaple in this route:
https://example.com/movie/[id]/[name]
the correct URL is: **https://example.com/movie/1/god-father**
but user enters wrong URL, for example: **https://example.com/movie/1/god**
So What should we do?
By default, nextJS will build this page. but we don't want to build it, because of wrong name! we want to redirect it to correct URL.
There should be a solution for this, because for each wrong URL **NextJS will build a useless file in disk**.
Thank for your time.

**Comment by brianlovin** [comment]
It will build a page, but the page logic itself can know whether data exists for `/movie/1/god` – if data exists, render! If not, show a 404 or handle a redirect based on whatever logic you have for correcting bad URLs.

**Reply by MohammadPCh** [reply]
Thanks for your reply but how can i redirect to correct URL in getStaticProps?

**Reply by brianlovin** [reply]
I don't think you can redirect in getStaticProps – that happens at build time. There was a good discussions about this though, here: https://github.com/zeit/next.js/discussions/11346 for more info. I've been seeing pretty consistent requests for better redirect logic in either `getStaticProps` and `getServerSideProps`, so maybe we could expect a better first-class solution to this down the road.

**Reply by MohammadPCh** [reply]
With this situation spammers can full our server disk easily.
just request like this:
https://example.com/movie/1/a1
to
https://example.com/movie/1/a1000000000000000000
and nextJs will build a page for each request!
Am i wrong?

**Reply by brianlovin** [reply]
This is a good question, I don't know what kind of abuse prevention mechanisms are in place for abusing static page creation. @Timer do you have any thoughts on this?

**Reply by Timer** [reply]
We're exploring a first class "404" return type from `getStaticProps` that'll skip storing the file.

**Reply by MohammadPCh** [reply]
> We're exploring a first class "404" return type from `getStaticProps` that'll skip storing the file.

So please make this unanswered. thanks.

**Comment by MohammadPCh** [comment]
Any update?

**Comment by MohammadPCh** [comment]
Dear @brianlovin ,
Can u find any solution? It's really a **security issue** and can effect on all NextJS websites that use incremental building!

**Reply by brianlovin** [reply]
I do not know the answer to this, sorry!

**Comment by yudazvi** [comment]
what about setting the preview mode to true for bad requests ?
